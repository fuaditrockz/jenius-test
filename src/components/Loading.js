import { Container } from "@chakra-ui/react";
import HashLoader from "react-spinners/HashLoader";

function Loading() {
  return (
    <Container style={{ marginBottom: 50, marginTop: 20 }}>
      <HashLoader
        color={"#FF8E53"}
        loading={true}
        size={150}
        aria-label="Loading Spinner"
        data-testid="loader"
        cssOverride={{
          display: "block",
          margin: "0 auto",
          borderColor: "red",
        }}
      />
    </Container>
  );
}

export default Loading;
