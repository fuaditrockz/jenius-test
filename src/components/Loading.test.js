import renderer from "react-test-renderer";
import Loading from "./Loading";

it("create component snapshot", () => {
  const component = renderer.create(<Loading />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
