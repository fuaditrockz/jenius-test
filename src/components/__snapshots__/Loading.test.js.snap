// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create component snapshot 1`] = `
<div
  className="chakra-container css-0"
  style={
    Object {
      "marginBottom": 50,
      "marginTop": 20,
    }
  }
>
  <span
    aria-label="Loading Spinner"
    data-testid="loader"
    style={
      Object {
        "borderColor": "red",
        "display": "block",
        "height": "150px",
        "margin": "0 auto",
        "position": "relative",
        "transform": "rotate(165deg)",
        "width": "150px",
      }
    }
  >
    <span
      style={
        Object {
          "animation": "react-spinners-HashLoader-before 2s infinite",
          "animationFillMode": "none",
          "borderRadius": "15px",
          "display": "block",
          "height": "30px",
          "left": "50%",
          "position": "absolute",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": "30px",
        }
      }
    />
    <span
      style={
        Object {
          "animation": "react-spinners-HashLoader-after 2s infinite",
          "animationFillMode": "none",
          "borderRadius": "15px",
          "display": "block",
          "height": "30px",
          "left": "50%",
          "position": "absolute",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": "30px",
        }
      }
    />
  </span>
</div>
`;
